### PR TITLE
Added Logger, Element class, SortTest class

### PR DESCRIPTION
**Pull request by ekaminskaya**
Created SortTest class
Created Element class

### CODE REVIEW

File: src/main/java/Element.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+import java.util.List;
+
+public class Element implements WebElement {
+
+    protected WebDriver driver = BaseDriver.initDriver();
+    protected WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(20));
+    protected Element element;
```
Comment (Lysenok) [inline src/main/java/Element.java]: Please use WebElement element; without modifications, and create a public metod with name of your class, where you could use in parameters WebElement element and using **this** show it at methods.

File: src/main/java/HomePage.java [hunk 2]
```diff
@@ -1,48 +1,75 @@
...
 
-public class HomePage extends BasePage{
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+public class HomePage extends BasePage {
+
+    private static final Logger log = LogManager.getLogger(HomePage.class);
```
Comment (Lysenok) [inline src/main/java/HomePage.java]: make it public and create metods info, error, debug where you could use in parameters String text.
your created log show at methods with parameters.

File: src/main/java/Element.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    public Element findMyElement(By locator) {
+        return (Element) driver.findElement(locator);
+    }
+
+    @Override
+    public void click() {
+        System.out.println("Click on the element");
```
Comment (qwertypp) [inline src/main/java/Element.java]: This click does nothing. Please fix also other methods

File: src/main/java/HomePage.java (hunk 2 above)
Comment (qwertypp) [reply src/main/java/HomePage.java]: Yes, good point. There are two possible solutions
1. Make it public static and import in all needed classes
2. Create an instance in BasePage and use it across all of the pages

File: src/main/java/HomePage.java [hunk 4]
```diff
@@ -1,48 +1,75 @@
...
+        log.info("Open the 'Home' page");
+        super.openPage(URL);
         if (!isElementPresent(navMenu)) {
             driver.close();
-            driver.get(baseUrl);
+            driver.get(BASE_URL);
         }
     }
 
+    public List<String> getSearchDropdownItems() {
```
Comment (qwertypp) [inline src/main/java/HomePage.java]: getSearchDropdownItemsString

File: src/main/java/SignUpPage.java [hunk 5]
```diff
@@ -16,42 +19,56 @@ public class SignUpPage extends BasePage {
     public By needHelpLink = By.xpath("//*[@class='a-expander-prompt']/ancestor::a");
 
     protected void openPage() {
-        super.openPage(url);
+        log.info("Open the 'SignUp' page");
+        super.openPage(URL);
     }
 
     public void fillUserName(String userName) {
+        log.info("Fill 'userName' field");
```
Comment (qwertypp) [inline src/main/java/SignUpPage.java]: those are fine but for more detailed logs you can use log.info("Fill 'userName' field with "+userName);

File: src/main/java/SignUpPage.java [hunk 6]
```diff
@@ -16,42 +19,56 @@ public class SignUpPage extends BasePage {
...
         driver.findElement(forgotPasswordLink).click();
+
     }
 
     public String getForgotPasswordFormHeader() {
+        log.info("Open 'Forgot password' page");
         return driver.findElement(forgotPasswordFormHeader).getText();
     }
 
+    public void fillPasswordField2(String password) {
```
Comment (qwertypp) [inline src/main/java/SignUpPage.java]: what 2 means?

File: src/test/java/SortTest.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import org.testng.annotations.Test;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class SortTest {
+
+    public List<String> sortDropdownItemsList(Collection collection) {
+        List<String> listForSorting = new ArrayList<>(collection);
+        Collections.sort(listForSorting, Collections.reverseOrder());
```
Comment (qwertypp) [inline src/test/java/SortTest.java:11]: this class is not important for review, but why are you using reverseOrder?

**Review by qwertypp — CHANGES_REQUESTED** [review]

File: src/test/java/SortTest.java (hunk 7 above)
Comment (ekaminskaya) [reply src/test/java/SortTest.java:11]: Used it just for practice comparator actions

File: src/main/java/HomePage.java (hunk 2 above)
Comment (ekaminskaya) [reply src/main/java/HomePage.java]: I'm still not clear why to make these changes.
1) Now I see a specific page in logs - like this [INFO ] 2022-10-14 18:55:48.970 [main] **LanguagePage** - Find default item in the 'Currency' list
Do I understand correctly that after changes I will see 'Bage page' only?
2) Why should I use other methods with parameters as String text if now I'm using the same log.info("text")

Comment (qwertypp) [reply src/main/java/HomePage.java]: 1. You could try and check
2. Sorry didn't get your question

**Pull request merged**